### ISSUE TITLE
UseDataClass triggers on value classes

### ISSUE THREAD

**Issue by ephemient** [detekt/detekt#4010]
## Expected Behavior

Detekt does not report "UseDataClass" on value classes.

## Observed Behavior

Detekt reports "UseDataClass" on value classes.

## Steps to Reproduce

```kotlin
@JvmInline
value class PlayerId(val value: Long)
```

## Context

Kotlin 1.5 renamed `inline class` to `value class`. The former was handled in #1450 but the latter is the stable feature.

## Your Environment

* Version of detekt used: 1.17.1
* Version of Gradle used (if applicable):
* Gradle scan link (add `--scan` option when running the gradle task):
* Operating System and version:
* Link to your project (if it's a public repository):
